cgi-bin/DW/Controller/Admin/SendMail.pm

```diff
@@ -171,5 +171,122 @@ sub index_controller {
     return DW::Template->render_template( 'admin/sendmail/index.tt', $rv );
 }
 
+sub lookup_controller {
+    my ( $ok, $rv ) = controller( privcheck => \@privs, form_auth => 1 );
+    return $rv unless $ok;
+    my $remote = $rv->{remote};
+
+    my $scope = sub { return '/admin/sendmail/lookup.tt' . $_[0] };
+
+    my $enhance_row = sub {
+        my ( $row ) = @_;
+        # link to view full message
+        $row->{msgurl} = '?id=' . $row->{msgid};
+        # time_sent needs a display version
+        $row->{time_sent_view} = LJ::time_to_http( $row->{time_sent} );
+        # build a request URL
+        if ( $row->{request} ) {
+            $row->{request_url} = "/support/see_request?id=";
+            $row->{request_url} .= $row->{request};
+        }
+        # sendto might be a uid
+        if ( $row->{sendto} =~ /^\d+$/ ) {
+            if ( my $u = LJ::load_userid( $row->{sendto} ) ) {
+                $row->{sendto} = $u;
+            } else {
+                $row->{sendto} = '[unknown user] '. $row->{sendto};
+            }
+        }
+        # sentfrom is definitely a uid
+        if ( my $ru = LJ::load_userid( $row->{remoteid} ) ) {
+            $row->{remote} = $ru;
+        } else {
+            $row->{remote} = '[unknown user] '. $row->{remoteid};
+        }
+        # add domain name to display site address
+        $row->{account_view} = $row->{account} . "\@$LJ::DOMAIN";
+
+        return $row;
+    };
+
+    my $r = DW::Request->get;
+    my $errors = DW::FormErrors->new;
+
+    if ( $r->did_post ) {
+        my $args = $r->post_args;
+
+        my $account = LJ::trim( $args->{account} );
+
+        if ( $account ) {
+            if ( my $priv = $LJ::SENDMAIL_ACCOUNTS{ $account } ) {
+                $errors->add( "account", ".error.nopriv", { priv => $priv } )
+                    unless $remote->has_priv( $priv );
+            } else {
+                $errors->add( "account", ".error.badacct" );
+            }
+
+        } else {
+            $errors->add( "account", ".error.noacct" );
+        }
+
+        unless ( $errors->exist ) {
+            my $dbr = LJ::get_db_reader();
+            my $rows = $dbr->selectall_hashref(
+                "SELECT msgid, time_sent, sendto, subject, request" .
+                " FROM siteadmin_email_history WHERE account=?", 'msgid',
+                undef, $account );
+            die $dbr->errstr if $dbr->err;
+
+            my @data = map { $enhance_row->( $_ ) } values %$rows;
+
+            $rv->{rows} = [ sort { $b->{time_sent} <=> $a->{time_sent} } @data ];
+            $rv->{account}  = $account;
+        }
+
+    } else {
+        my $args = $r->get_args;
```

Comment:
I think it's clearer the other way -- get vs post is, for me, much useful if the get shows a form that then does a post. 




> On Feb 9, 2015, at 11:50, Jen <notifications@github.com> wrote:
> 
> In cgi-bin/DW/Controller/Admin/SendMail.pm:
> 
> > +        unless ( $errors->exist ) {
> > +            my $dbr = LJ::get_db_reader();
> > +            my $rows = $dbr->selectall_hashref(
> > +                "SELECT msgid, time_sent, sendto, subject, request" .
> > +                " FROM siteadmin_email_history WHERE account=?", 'msgid',
> > +                undef, $account );
> > +            die $dbr->errstr if $dbr->err;
> > +
> > +            my @data = map { $enhance_row->( $_ ) } values %$rows;
> > +
> > +            $rv->{rows} = [ sort { $b->{time_sent} <=> $a->{time_sent} } @data ];
> > +            $rv->{account}  = $account;
> > +        }
> > +
> > +    } else {
> > +        my $args = $r->get_args;
> I thought it would be OK to have the behavior be different for a GET vs. a POST but I can look into that.
> 
> —
> Reply to this email directly or view it on GitHub.
>